src/vs/editor/common/config/commonEditorConfig.ts

```diff
@@ -458,6 +458,7 @@ export abstract class CommonEditorConfiguration extends Disposable implements IC
 
 	protected abstract readConfiguration(styling: BareFontInfo): FontInfo;
 
+	public abstract reserveHeight(height: number): void;
```

Comment:
Now I realize that this method could be implemented here because it doesn't need any dom stuff (it can live in `/common/` here). `_computeInternalOptions` could then subtract the reserved height.